### DISCUSSION TITLE
Fresh 2.3 install, no alerts showing

### DISCUSSION THREAD

**Discussion by paulgear** [Security-Onion-Solutions/securityonion#4327] [Unsupported Versions] [answered]
Hi folks,
I've just done a fresh install of my SO standalone box on 2.3.  All OS and SO updates have been applied, and my span port configuration hasn't changed, and I can see packets on the interface if I sniff with tshark, but I'm not getting anything at all in the alerts.  Status looks fine:

    Checking Docker status

    Docker --------------------------------------------------------- [ OK ]    

    Checking container statuses

    so-aptcacherng ------------------------------------------------- [ OK ]    
    so-curator ----------------------------------------------------- [ OK ]    
    so-dockerregistry ---------------------------------------------- [ OK ]    
    so-elastalert -------------------------------------------------- [ OK ]    
    so-elasticsearch ----------------------------------------------- [ OK ]    
    so-filebeat ---------------------------------------------------- [ OK ]    
    so-fleet ------------------------------------------------------- [ OK ]    
    so-grafana ----------------------------------------------------- [ OK ]    
    so-idstools ---------------------------------------------------- [ OK ]    
    so-influxdb ---------------------------------------------------- [ OK ]    
    so-kibana ------------------------------------------------------ [ OK ]    
    so-kratos ------------------------------------------------------ [ OK ]    
    so-logstash ---------------------------------------------------- [ OK ]    
    so-mysql ------------------------------------------------------- [ OK ]    
    so-nginx ------------------------------------------------------- [ OK ]    
    so-redis ------------------------------------------------------- [ OK ]    
    so-sensoroni --------------------------------------------------- [ OK ]    
    so-soc --------------------------------------------------------- [ OK ]    
    so-soctopus ---------------------------------------------------- [ OK ]    
    so-steno ------------------------------------------------------- [ OK ]    
    so-strelka-backend --------------------------------------------- [ OK ]    
    so-strelka-coordinator ----------------------------------------- [ OK ]    
    so-strelka-filestream ------------------------------------------ [ OK ]    
    so-strelka-frontend -------------------------------------------- [ OK ]    
    so-strelka-gatekeeper ------------------------------------------ [ OK ]    
    so-strelka-manager --------------------------------------------- [ OK ]    
    so-suricata ---------------------------------------------------- [ OK ]    
    so-telegraf ---------------------------------------------------- [ OK ]    
    so-zeek -------------------------------------------------------- [ OK ]    

There doesn't seem to be any advice in the 2.3 docs about troubleshooting alerts.  Have I missed something?  What's the next thing to try?

**Comment by mbaki** [comment, marked as answer]
I am using emerging threats Pro (Suricata), and under /opt/so/conf/suricata/rules/all.rules, enable:

alert ip any any -> any any (msg:"GPL ATTACK_RESPONSE id check returned root"; content:"uid=0|28|root|29|"; classtype:bad-unknown; sid:2100498; rev:7; metadata:created_at 2010_09_23, updated_at 2010_09_23;)

Then test by browse or curl:
https://testmynids.org/uid/index.html

Hope this helps

**Reply by paulgear** [reply]
That rule was already enabled, as far as I can tell, and the test site did not cause any alerts.  So I guess it didn't help.

But I started poking around in the Suricata container and found it was listening on bond0.  Looking at the output of networkctl, it says that eno2 (the interface that is supposed to be in the bond) is "degraded" and "configuring".  I found /etc/NetworkManager/system-connections/bond0{,-slave-eno2}, and between networkctl, nmcli, and netplan, I determined that netplan was managing eno2 (it was in the file created by the base Ubuntu install), which was preventing NetworkManager form managing it, which was preventing from it setting up the bond.  So I guess it did help.

As an aside, I'm a bit surprised that Security Onion is moving towards Network Manager for managing network connections when it seems like the Linux community overall is moving away from it.

**Reply by paulgear** [reply]
Even after solving the above issue with bond0, I still have no alerts.  so-steno is gathering data and dumping it into /nsm/pcap, but so-suricata reports the following:

```
root@so:~# docker logs so-suricata
29/5/2021 -- 01:01:47 - <Notice> - This is Suricata version 6.0.2 RELEASE running in SYSTEM mode
29/5/2021 -- 01:02:20 - <Error> - [ERRCODE: SC_ERR_INVALID_ARGUMENT(13)] - worker-cpu-set: upper bound (2) of cpu set is too high, only 2 cpu(s)
29/5/2021 -- 01:02:20 - <Notice> - all 1 packet processing threads, 4 management threads initialized, engine started.
29/5/2021 -- 01:02:50 - <Error> - [ERRCODE: SC_ERR_AFP_READ(191)] - Error reading data from iface 'bond0': (100) Network is down
```

The network interface is definitely not down:
```
root@so:~# docker exec -ti so-suricata /bin/bash
[root@so /]# ip addr ls dev bond0
71: bond0: <BROADCAST,MULTICAST,PROMISC,MASTER,UP,LOWER_UP> mtu 9000 qdisc noqueue state UP group default qlen 1000
    link/ether 3c:a8:2a:a0:29:05 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::b4c2:7205:fda7:7954/64 scope link noprefixroute 
       valid_lft forever preferred_lft forever
```

If I install tcpdump or wireshark in the so-suricata container, I can dump traffic from bond0 successfully.

Is Suricata not seeing traffic on the interface the reason for my lack of alerts, or is there something else wrong?

**Reply by weslambert** [reply]
What is the output of nmcli con show?  Have you tried simply rebooting?

**Reply by paulgear** [reply]
Apologies, I should have updated this topic earlier.  I found that the NetworkManager configuration was trying to configure an IP address on the sniffer interface, and when it failed, it cycled the link down & up again, so the interface was actually going down from Suricata's perspective, but I was only seeing it during the times when it was up (and searching for an IP).  Reconfiguring the interface not to require an IPv4 or IPv6 address solved the problem and I'm now seeing Suricata alerts.

**Reply by TotieBash** [reply]
I'm getting the same isssue where /nsm/suricata/eve json files are empty..  This is a fresh install of SO 2.3.52 from iso build.  @paulgear can you pls elaborate the fix?  Thanks

**Reply by frankofamerica-net** [reply]
> Apologies, I should have updated this topic earlier. I found that the NetworkManager configuration was trying to configure an IP address on the sniffer interface, and when it failed, it cycled the link down & up again, so the interface was actually going down from Suricata's perspective, but I was only seeing it during the times when it was up (and searching for an IP). Reconfiguring the interface not to require an IPv4 or IPv6 address solved the problem and I'm now seeing Suricata alerts.

I am experiencing the same issue(s) with NIC. Could you elaborate on how you "reconfigured the interface not to require an IPv4 or IPv6 address solved the problem..."?

**Reply by dougburks** [reply]
@frankofamerica-net Please review https://docs.securityonion.net/en/2.3/suricata.html#troubleshooting-alerts. If you have further questions or problems, please start a new discussion as mentioned at https://github.com/Security-Onion-Solutions/securityonion/discussions/1720.
